reana_commons/utils.py

```diff
@@ -213,6 +213,9 @@ def get_disk_usage(directory, summarize=False, to_human_readable_units=None):
         command.append("-s")
     else:
         command.append("-a")
+    if not "Darwin" in platform.system():
+        # Default block size in GNU is KB
+        command.append("-b")
```

Comment:
Works better, two comments:

**(1)** Are totals fine?  E.g. I get:

```
$ reana-client quota-show --resource disk
333.57 KB used
```

and:

```
$ du -hsc /var/reana/users/00000000-0000-0000-0000-000000000000/workflows
408K   /var/reana/users/00000000-0000-0000-0000-000000000000/workflows
408K    total
```

and in the UI:

![2020-11-04-145320_724x338_scrot](https://user-images.githubusercontent.com/517546/98119867-7ec5cd80-1ead-11eb-8aa7-05a48bd4988a.png)

which would naively be 309.57K.

So it looks like we have three different numbers, as it were... Are we calculating KB and KiB everywhere in the same manner?

**(2)** The release commit is perhaps coming too soon, as there are other PRs for `r-commons`, one of them already approved. We can merge everything pending and create the release commit only afterwards?

Reply:
**(1)**

My suspicion is that this is due to the fact that `du` always shows blocks of at least 4K to represent sizes in KB. See for instance:
```
$ du -s /var/reana/users/00000000-0000-0000-0000-000000000000/
292     /var/reana/users/00000000-0000-0000-0000-000000000000/
```

vs

```
$ du -s -b /var/reana/users/00000000-0000-0000-0000-000000000000/
227398  /var/reana/users/00000000-0000-0000-0000-000000000000/
```

227398 Bytes = 222.068359375 KB != 292K

We **always** grab the value in bytes, so there are these differences in precisions. Do you know any solution?

**(2)**

Done.

Reply:
Starting from bytes on my system:

```console
$ du -sb /var/reana/users/00000000-0000-0000-0000-000000000000
345669  /var/reana/users/00000000-0000-0000-0000-000000000000
```

gives: 345669 bytes = 345.669 KB = 337.56 KiB... while we display 333.57 KB usage.

This means that the total number that we display is much closer to KiB units rather than KB units, as I suspected might be the case... Do we calculate KB/KiB always right? Or are we simply displaying wrong units?

To confirm/infirm this, I have made a quick helloworld test with a bigger extra data file:

```
$ ls -l bigfile
-rw-r--r-- 1 simko simko 838860800 Nov  4 18:56 bigfile
$ ls -lh bigfile
-rw-r--r-- 1 simko simko 800M Nov  4 18:56 bigfile
$ du -s -h /var/reana/users/00000000-0000-0000-0000-000000000000/workflows/4b40df8f-b776-416b-be42-e1f845fa0e68
801M    /var/reana/users/00000000-0000-0000-0000-000000000000/workflows/4b40df8f-b776-416b-be42-e1f845fa0e68
$ du -s --si /var/reana/users/00000000-0000-0000-0000-000000000000/workflows/4b40df8f-b776-416b-be42-e1f845fa0e68
839M    /var/reana/users/00000000-0000-0000-0000-000000000000/workflows/4b40df8f-b776-416b-be42-e1f845fa0e68
$ du -s -b /var/reana/users/00000000-0000-0000-0000-000000000000/workflows/4b40df8f-b776-416b-be42-e1f845fa0e68
838880491       /var/reana/users/00000000-0000-0000-0000-000000000000/workflows/4b40df8f-b776-416b-be42-e1f845fa0e68
```

... which means 838880491 bytes = 838.880491 MB = 800.02 MiB, while REANA displays 800.02 MB instead:

![2020-11-04-190358_710x107_scrot](https://user-images.githubusercontent.com/517546/98151472-8185ea00-1ed0-11eb-877c-6e75462d2faa.png)

... which seems to confirm the hypothesis.

There are two solutions in my eyes:

1. Keep displaying the number we have (800.02), but change units to KiB, MiB for both CLI and Web clients.
2. Keep displaying KB, MB units, but then with the proper 839M number.

This is going a bit further beyond the given PR at hand, but it would be good to complement the `-b` fix with fixing also units, so that we can fully close the original issue.

Reply:
Discussed offline to go for solution **2.**

Implemented here: https://github.com/reanahub/reana-db/pull/111

Reply:
Can you please rebase against latest `master`?

Reply:
Wow, I can't see that the branch is out-of-date.